File: jung-visualization/src/main/java/edu/uci/ics/jung/visualization/spatial/rtree/SplitterContext.java
package edu.uci.ics.jung.visualization.spatial.rtree;

/**
 * a container for the functions that support R-Tree and R*-Tree
 *
 * @param <T> the type of element in the RTree
 * @author Tom Nelson
 */
public class SplitterContext<T> {

  public final LeafSplitter<T> leafSplitter;
  public final Splitter<T> splitter;

  public static <T> SplitterContext<T> of(LeafSplitter<T> leafSplitter, Splitter<T> splitter) {
    return new SplitterContext<>(leafSplitter, splitter);
  }

  private SplitterContext(LeafSplitter<T> leafSplitter, Splitter<T> splitter) {
    this.leafSplitter = leafSplitter;
    this.splitter = splitter;
  }
}

File: jung-visualization/src/main/kotlin/edu/uci/ics/jung/visualization/spatial/rtree/SplitterContext.kt
package edu.uci.ics.jung.visualization.spatial.rtree

/**
 * a container for the functions that support R-Tree and R*-Tree
 *
 * @param T the type of element in the RTree
 * @author Tom Nelson
 */
class SplitterContext<T> private constructor(
  @JvmField val leafSplitter: LeafSplitter<T>,
  @JvmField val splitter: Splitter<T>
) {
  companion object {
    @JvmStatic
    fun <T> of(leafSplitter: LeafSplitter<T>, splitter: Splitter<T>): SplitterContext<T> =
      SplitterContext(leafSplitter, splitter)
  }
}
